app/Http/Controllers/AtcBookingController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\AtcBooking;
+use Auth;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+
+class AtcBookingController extends Controller {
+    public function viewBookings() {
+        $today = Carbon::today();
+        $bookings = AtcBooking::whereDate('start', '>=', $today)
+            ->where(function ($query) use ($today) {
+                $query->whereDate('end', '<=', $today->addDays(14))
+                      ->orWhere('cid', Auth::id());
+            })
+            ->orderBy('start', 'ASC')
+            ->get();
+
+        $bookings = groupAtcBookingsByDate($bookings);
+
+        $types = [
+            AtcBooking::TYPES['BOOKING'] => "Booking"
+        ];
+
+        if (Auth::user()->isAbleTo('train') || Auth::user()->isAbleTo('snrStaff')) {
+            $types[AtcBooking::TYPES['MONITORING']] = "Monitoring";
+        }
+
+        if (Auth::user()->isAbleTo('events') || Auth::user()->isAbleTo('snrStaff')) {
```

Comment:
Should also allow this trait here: `Auth::user()->hasRole('events-team')`

Reply:
Resolved